### PR TITLE
pdfid: init at 0.2.8

### PR DESCRIPTION
**Pull request by D3vil0p3r**
## Things done

- Built on platform(s)
  - [X] x86_64-linux
  - [ ] aarch64-linux
  - [ ] x86_64-darwin
  - [ ] aarch64-darwin
- For non-Linux: Is sandboxing enabled in `nix.conf`? (See [Nix manual](https://nixos.org/manual/nix/stable/command-ref/conf-file.html))
  - [ ] `sandbox = relaxed`
  - [ ] `sandbox = true`
- [X] Tested, as applicable:
  - [NixOS test(s)](https://nixos.org/manual/nixos/unstable/index.html#sec-nixos-tests) (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
  - and/or [package tests](https://nixos.org/manual/nixpkgs/unstable/#sec-package-tests)
  - or, for functions and "core" functionality, tests in [lib/tests](https://github.com/NixOS/nixpkgs/blob/master/lib/tests) or [pkgs/test](https://github.com/NixOS/nixpkgs/blob/master/pkgs/test)
  - made sure NixOS tests are [linked](https://nixos.org/manual/nixpkgs/unstable/#ssec-nixos-tests-linking) to the relevant packages
- [ ] Tested compilation of all packages that depend on this change using `nix-shell -p nixpkgs-review --run "nixpkgs-review rev HEAD"`. Note: all changes have to be committed, also see [nixpkgs-review usage](https://github.com/Mic92/nixpkgs-review#usage)
- [X] Tested basic functionality of all binary files (usually in `./result/bin/`)
- [24.05 Release Notes](https://github.com/NixOS/nixpkgs/blob/master/nixos/doc/manual/release-notes/rl-2405.section.md) (or backporting [23.05](https://github.com/NixOS/nixpkgs/blob/master/nixos/doc/manual/release-notes/rl-2305.section.md) and [23.11](https://github.com/NixOS/nixpkgs/blob/master/nixos/doc/manual/release-notes/rl-2311.section.md) Release notes)
  - [ ] (Package updates) Added a release notes entry if the change is major or breaking
  - [ ] (Module updates) Added a release notes entry if the change is significant
  - [ ] (Module addition) Added a release notes entry if adding a new NixOS module
- [X] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md).

---

Add a :+1: [reaction] to [pull requests you find important].

[reaction]: https://github.blog/2016-03-10-add-reactions-to-pull-requests-issues-and-comments/
[pull requests you find important]: https://github.com/NixOS/nixpkgs/pulls?q=is%3Aopen+sort%3Areactions-%2B1-desc

### CODE REVIEW

File: pkgs/by-name/pd/pdfid/package.nix [hunk 1]
```diff
@@ -0,0 +1,40 @@
...
+
+python3Packages.buildPythonApplication rec {
+  pname = "pdfid";
+  version = "0.2.8";
+  format = "other";
+
+  src = fetchzip {
+    url = "https://didierstevens.com/files/software/pdfid_v0_2_8.zip";
+    hash = "sha256-ZLyhBMF2KMX0c1oCvuSCjEjHTnm2gFhJtasaTD9Q1BI=";
+    stripRoot=false;
```
Comment (drupol) [inline pkgs/by-name/pd/pdfid/package.nix:16]:
```suggestion
    stripRoot = false;
```

Consistency

File: pkgs/by-name/pd/pdfid/package.nix [hunk 2]
```diff
@@ -0,0 +1,40 @@
...
+  };
+
+  nativeBuildInputs = [
+    makeWrapper
```
Comment (drupol) [inline pkgs/by-name/pd/pdfid/package.nix:20]: I would use `makeBinaryWrapper` instead, it has a better compatibility with Darwin platforms.

File: pkgs/by-name/pd/pdfid/package.nix [hunk 3]
```diff
@@ -0,0 +1,40 @@
...
+  ];
+
+  installPhase = ''
+    runHook preInstall
+    mkdir -p $out/{bin,share/pdfid}
+    cp -a * $out/share/pdfid/
+    makeWrapper ${python3}/bin/python $out/bin/pdfid \
```
Comment (drupol) [inline pkgs/by-name/pd/pdfid/package.nix:27]:
```suggestion
    makeWrapper ${lib.getExe python3} $out/bin/${meta.mainProgram} \
```

File: pkgs/by-name/pd/pdfid/package.nix [hunk 4]
```diff
@@ -0,0 +1,40 @@
...
+  '';
+
+  meta = with lib; {
+    description = "Scan a file to look for certain PDF keywords";
+    mainProgram = "pdfid";
+    homepage = "https://blog.didierstevens.com/programs/pdf-tools/";
+    maintainers = with maintainers; [ d3vil0p3r ];
+    platforms = platforms.unix;
+    license = with licenses; [ free ];
+  };
```
Comment (drupol) [inline pkgs/by-name/pd/pdfid/package.nix:39]: Should be sorted alphabetically. Consistency is key.

**Review by drupol — CHANGES_REQUESTED** [review]
Hi !

Thanks for your contribution.

I've left some comments, let me know if this is clear enough.

**Comment by D3vil0p3r** [comment]
@drupol thank you very much for your suggestions.

**Review by drupol — APPROVED** [review]

**Pull request merged**